sandbox/src/game/interface/Tooltip.cpp

```diff
@@ -0,0 +1,120 @@
+#include "Tooltip.h"
+
+void Tooltip::DrawBackground()
+{
+	Square::Vector2 position = mTitle->Pos();
+	int startX = position.x - MaxWidth() / 2;
+	int endX = position.x + MaxWidth() / 2;
+	int startY = position.y - mTitle->ScaledDimensions().y / 2 - PADDING;
+	int endY = startY + TotalHeight() + PADDING;
+
+	for (int x = startX; x < endX; x++)
+	{
+		for (int y = startY; y < endY; y++)
+		{
+			bool border = x == startX || y == startY || x == endX - 1 || y == endY - 1;
+
+			if (border)
+				mGraphics.DrawPixel(Square::Vector2(x, y), { 0, 0, 0, 255 });
+			else
+				mGraphics.DrawPixel(Square::Vector2(x, y), { 230, 230, 230, 255 });
+		}
+	}
+}
+
+void Tooltip::DrawTitleBackground()
+{
+	int startX = mTitle->Pos().x - mTitle->ScaledDimensions().x / 2 - PADDING/2;
+	int endX = mTitle->Pos().x + mTitle->ScaledDimensions().x / 2 + PADDING/2;
+	int startY = mTitle->Pos().y - mTitle->ScaledDimensions().y / 2 - PADDING/2;
+	int endY = mTitle->Pos().y + mTitle->ScaledDimensions().y / 2 + PADDING/2;
+
+	for (int x = startX + 5; x < endX - 5; x++)
+	{
+		for (int y = startY; y < endY; y++)
+		{
+			mGraphics.DrawPixel(Square::Vector2(x, y), { 0, 0, 0, 255 });
+		}
+	}
+}
+
+int Tooltip::TotalHeight()
+{
+	int height = mTitle->ScaledDimensions().y;
+	if (!mButtons.empty())
+		height += (mButtons.back()->Position().y - mTitle->Pos().y) + mButtons.back()->Height() / 2;
+
+	return height;
+}
+
+int Tooltip::MaxWidth()
+{
+	int maxButtonWidth = 0;
+	for (auto button : mButtons)
+		maxButtonWidth = std::max(button->Width(), maxButtonWidth);
+
+	return std::max((int) mTitle->ScaledDimensions().x + PADDING * 2, maxButtonWidth);
+}
+
+Tooltip::Tooltip(const std::string text, std::vector<std::string> commands)
```

Comment:
Place constructors and desctructors at the top of the cpp file. I know visual studio doesn't do that by default but it's really where they need to be

Reply:
It should be ```const std::string& text```